graphics/macgui/mactext-canvas.cpp

```diff
@@ -82,9 +83,17 @@ void MacTextCanvas::chopChunk(const Common::U32String &str, int *curLinePtr, int
 
 	chunk->getFont()->wordWrapText(str, maxWidth, text, lineContinuations, w);
 
-	if (text.size() == 0) {
+	for (int i = 0; i < (int)text.size(); i++) {
+		D(9, "Line Continuations [%d] : %d", i, lineContinuations[i]);
+	}
+
+	if (text.empty()) {
 		D(5, "chopChunk: too narrow width, >%d", maxWidth);
-		chunk->text += str;
+
+		if (w < maxWidth) {
+			chunk->text += str;	//Only append if within bounds
```

Comment:
```suggestion
			chunk->text += str;	// Only append if within bounds
```

And what is happening otherwise, e.g when not within bounds?

Reply:
This means even if the wrapping failed, if there's still room (w < maxWidth), it tries to append the text.
If not ,the string is discarded silently (does not get appended if already out of bounds).